opencog/atoms/base/Atom.h

```diff
@@ -190,6 +195,16 @@ class Atom
         return classserver().isA(at, t);
     }
 
+    /** Returns the outgoing set.
+     *
+     * @return The outgoing set.
+     */
+    virtual inline const HandleSeq& getOutgoingSet() const
+    {
+        static HandleSeq no_atoms;
+        return no_atoms;
+    }
```

Comment:
Wow. I just did a disassembly of the code for this:
```
char * foo(const Handle& h) {
   NodePtr p(NodeCast(h));
   if (p) return "asdf";
   return nullptr;
}
```
and it is insanely, jaw-droppingly horrific! I am just flabbergasted!  I could have sworn that last time I did this, it was nice and clean and elegant, so I'm a bit flummoxed at the moment.